docs/DeveloperGuide.md

```diff
@@ -6,33 +6,360 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### User Interface (UI) - Class Implementation
+![img.png](img.png) <br>
+**Figure 1: UML Diagram of User Interface (UI) Class**
+1. The **`UI class`** named `Ui` belongs to the package functionalities.ui. The class contains several static and non-static methods that display messages to the user such as **`showUserMessage()`**, **`showErrorMessage()`**, and **`showWelcomeMessage()`**. The class is also used to read user input, format and print appointment lists, and add or remove appointments from the **`sniff appointment manager`**. The **`Ui`** class has a **`showLine()`** method that displays a divider line to the user.
+2. The **`UI class`** has a private static final String DOT_THEN_SPACE field that is used as a constant string value to format user messages. It has a **`readUserCommand()`** method that returns a string value that is entered by the user through the Command Line Interface (CLI).
+3. The class imports the **`SniffException`** class from the exception package and utilizes it in the **`showErrorMessage()`** method to display an error message to the user.
 
+### Command - Class Implementation
+![img_16.png](Updated%20Command%20UML.png) <br>
```

Comment:
I am not very clear what the symbols represent in the class diagrams, is a different standard being used to represent them?